kalite/distributed/static/js/distributed/exercises/views.js

```diff
@@ -673,103 +580,201 @@ window.ExercisePracticeView = Backbone.View.extend({
             this.current_attempt_log.set({
                 complete: true
             });
+            if (this.correct_updates) {
+                this.correct_updates();
+            }
         }
 
         this.current_attempt_log.save();
 
     },
 
-    ready_for_next_question: function() {
+    update_total_points: function(data) {
+        // update the top-right point display, now that we've saved the points successfully
+        if (this.log_model.has("points")) {
+            window.statusModel.update_total_points(this.log_model.get("points") - this.status_points);
+            this.status_points = this.log_model.get("points");
+        }
+    },
 
-        var self = this;
+    get_points_per_question: function() {
+        return this.attempt_collection.calculate_points_per_question(this.exercise_view.data_model.get("basepoints"));
+    },
 
-        if (this.user_data_loaded_deferred) {
 
-            this.user_data_loaded_deferred.then(function() {
+    display_message: function() {
```

Comment:
I would like to see comments giving some context to these functions. For example, what kind of message does this display? How is it meant to be used by the developer? It's not immediately obvious how to follow the control flow to this function, which is reached by passing control from `ExerciseWrapperBaseView.initialize` -> `ExercisePracticeView.load_user_data` -> `ExercisePracticeView.user_data_loaded` -> `ExerciseWrapperBaseView.display_message`. It's also not used by any of the other descendants of `ExerciseWrapperBaseView`, which suggests it should be defined in `ExercisePracticeView`.